From fd62cacfc059913fa80034a3c5af67af2adebf3e Mon Sep 17 00:00:00 2001
From: Steve Kipisz <s-kipisz2@ti.com>
Date: Tue, 26 Feb 2013 14:32:05 -0600
Subject: [PATCH 4/4] am335x_evm:Fixup some configurations and environment variables

This patch adds configurations to boards.cfg
- Adds DFU_NAND to am335x_evm
- Adds am335x_evm_dfu_mmc

Adds dfu environment variables to am335x_evm.h.
Fixes some mmc environment variables in am335x_evm.h.

Signed-off-by: Steve Kipisz <s-kipisz2@ti.com>
---
 boards.cfg                   |    4 ++-
 include/configs/am335x_evm.h |   70 ++++++++++++++++++++++++++++++++---------
 2 files changed, 57 insertions(+), 17 deletions(-)

diff --git a/boards.cfg b/boards.cfg
index cb2dd51..e9ade26 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -230,7 +230,9 @@ versatileqemu                arm         arm926ejs   versatile           armltd
 integratorap_cm946es         arm         arm946es    integrator          armltd         -               integratorap:CM946ES
 integratorcp_cm946es         arm         arm946es    integrator          armltd         -               integratorcp:CM946ES
 ca9x4_ct_vxp                 arm         armv7       vexpress            armltd
-am335x_evm                   arm         armv7       am335x              ti             am33xx      am335x_evm:SERIAL1,CONS_INDEX=1
+am335x_evm                   arm         armv7       am335x              ti             am33xx      am335x_evm:SERIAL1,CONS_INDEX=1,DFU_NAND
+am335x_evmf_dfu_mmc          arm         armv7       am335x              ti             am33xx      am335x_evm:SERIAL1,CONS_INDEX=1,DFU_MMC
+am335x_evm_restore_flash     arm         armv7       am335x              ti             am33xx      am335x_evm:SERIAL1,CONS_INDEX=1,RESTORE_FLASH
 am335x_evm_spiboot           arm         armv7       am335x              ti             am33xx      am335x_evm:SERIAL1,CONS_INDEX=1,SPI_BOOT
 am335x_evm_uart1             arm         armv7       am335x              ti             am33xx      am335x_evm:SERIAL2,CONS_INDEX=2
 am335x_evm_uart2             arm         armv7       am335x              ti             am33xx      am335x_evm:SERIAL3,CONS_INDEX=3
diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index c4482df..a3ecdce 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -44,11 +44,43 @@
 /* commands to include */
 #include <config_cmd_default.h>
 
+#define CONFIG_CMD_MTDPARTS
+#define MTDIDS_DEFAULT		"nand0=omap2-nand.0"
+#define MTDPARTS_DEFAULT	 "mtdparts=omap2-nand.0:128k(SPL)," \
+				"128k(SPL.backup1),128k(SPL.backup2)," \
+				"128k(SPL.backup3),1920k(u-boot)," \
+				"128k(u-boot-env),5m(kernel),-(rootfs)" \
+
+/* USB Device Firmware Update (DFU) support */
+#define DFU_ALT_INFO_NAND \
+	"dfu_alt_info=" \
+	"SPL part 0 1;" \
+	"SPL.backup1 part 0 2;" \
+	"SPL.backup2 part 0 3;" \
+	"SPL.backup3 part 0 4;" \
+	"u-boot part 0 5;" \
+	"kernel part 0 7;" \
+	"rootfs part 0 8\0" \
+
+#define DFU_ALT_INFO_MMC \
+	"dfu_alt_info=" \
+	"boot part 0 1;" \
+	"rootfs part 0 2;" \
+	"MLO fat 0 1;" \
+	"u-boot.img fat 0 1;" \
+	"uEnv.txt fat 0 1\0"
+
+#ifdef CONFIG_DFU_MMC
+#define CONFIG_DFU_ALT                  DFU_ALT_INFO_MMC
+#elif CONFIG_DFU_NAND
+#define CONFIG_DFU_ALT                  DFU_ALT_INFO_NAND
+#else
+#define CONFIG_DFU_ALT
+#endif
+
 #define CONFIG_CMD_ASKENV
 #define CONFIG_VERSION_VARIABLE
 
-/* set to negative value for no autoboot */
-#define CONFIG_BOOTDELAY		1
 #define CONFIG_ENV_VARS_UBOOT_CONFIG
 #define CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
 #define CONFIG_EXTRA_ENV_SETTINGS \
@@ -75,8 +107,8 @@
 	"ramroot=/dev/ram0 rw ramdisk_size=65536 initrd=${rdaddr},64M\0" \
 	"ramrootfstype=ext2\0" \
 	"ip_method=none\0" \
-	"bootargs_default=setenv bootargs " \
-		"console=${console} "\
+	"bootargs_defaults=setenv bootargs " \
+		"console=${console} " \
 		"${optargs}\0" \
 	"mmcargs=run bootargs_defaults;" \
 		"setenv bootargs ${bootargs} " \
@@ -139,7 +171,11 @@
 			"setenv fdtfile am335x-evm.dtb; fi; " \
 		"if test $board_name = A335X_SK; then " \
 			"setenv fdtfile am335x-evmsk.dtb; fi\0" \
-	CONFIG_DFU_ALT
+	CONFIG_DFU_ALT 
+
+#ifndef CONFIG_RESTORE_FLASH
+/* set to negative value for no autoboot */
+#define CONFIG_BOOTDELAY	3
 
 #define CONFIG_BOOTCOMMAND \
 	"mmc dev ${mmcdev}; if mmc rescan; then " \
@@ -156,13 +192,24 @@
 			"run mmcboot;" \
 		"elif run loaduimage; then " \
 			"run mmcboot;" \
-		"else "\
+		"else " \
 			"echo Could not find ${bootfile} ;" \
 		"fi;" \
 	"else " \
 		"run nandboot;" \
 	"fi;" \
 
+#else
+#define CONFIG_BOOTDELAY	0
+
+#define CONFIG_BOOTCOMMAND
+	"setenv autoload no; "			\
+	"dhcp; "				\
+	"if tftp 80000000 debrick.scr; then "	\
+		"source 80000000; "		\
+	"fi"
+#endif
+
 /* Clock Defines */
 #define V_OSCK				24000000  /* Clock output from T2 */
 #define V_SCLK				(V_OSCK)
@@ -213,7 +260,6 @@
 #define CONFIG_USB_GADGET
 #define CONFIG_USBDOWNLOAD_GADGET
 #define CONFIG_DFU_FUNCTION
-#define CONFIG_DFU_MMC
 
 /* USB TI's IDs */
 #define CONFIG_USBD_HS
@@ -221,14 +267,6 @@
 #define CONFIG_G_DNL_PRODUCT_NUM 0xBD00
 #define CONFIG_G_DNL_MANUFACTURER "Texas Instruments"
 
-#define CONFIG_DFU_ALT \
-	"dfu_alt_info=" \
-	"boot part 0 1;" \
-	"rootfs part 0 2;" \
-	"MLO fat 0 1;" \
-	"u-boot.img fat 0 1;" \
-	"uEnv.txt fat 0 1\0"
-
 #define CONFIG_CMD_DFU
 
  /* Physical Memory Map */
@@ -413,7 +451,7 @@
 # define CONFIG_ENV_SECT_SIZE		(4 << 10) /* 4 KB sectors */
 #endif /* SPI support */
 
-#if defined(CONFIG_SPL_BUILD) && defined(CONFIG_SPL_USBETH_SUPPORT)
+#if defined(CONFIG_SPL_BUILD) || defined(CONFIG_SPL_USBETH_SUPPORT)
 /* disable host part of MUSB in SPL */
 #undef CONFIG_MUSB_HOST
 /*
-- 
1.7.2.3

